composer/blog/gcp-tech-blog/data-orchestration-with-composer/dags/data_orchestration_blog_sample_dag.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2021 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from airflow.utils import dates
+from airflow import models
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils.state import State
+from airflow.providers.google.cloud.sensors.gcs import GCSObjectExistenceSensor
+from airflow.providers.google.cloud.operators.dataflow import DataflowTemplatedJobStartOperator
+from airflow.providers.google.cloud.operators.gcs import GCSDeleteBucketOperator
+from airflow.providers.google.cloud.operators.bigquery import BigQueryCheckOperator
+from airflow.contrib.operators.slack_webhook_operator import SlackWebhookOperator
+
+BUCKET_NAME = models.Variable.get("bucket_name")
+FILE_NAME = models.Variable.get("file_name")
+BQ_SQL_STRING = models.Variable.get("bq_sql_string")
+
+def on_failure_callback(context):
+    ti = context.get('task_instance')
+    slack_msg = """
+            :red_circle: Task Failed. 
+            *Task*: {task}  
+            *Dag*: {dag} 
+            *Execution Time*: {exec_date}  
+            *Log Url*: {log_url} 
+            """.format(
+            task=ti.task_id,
+            dag=ti.dag_id,
+            log_url=ti.log_url,
+            exec_date=context.get('execution_date'),
+        )
+    slack_webhook_token = BaseHook.get_connection('slack_connection').password
+    slack_error = SlackWebhookOperator(
+        task_id='post_slack_error',
+        http_conn_id='slack_connection',
+        channel='#airflow-alerts',
+        webhook_token=slack_webhook_token,
+        message=slack_msg)
+    slack_error.execute(context)
+
+with models.DAG(
+    'transform-crm-workload',
+    schedule_interval=None,
+    start_date=dates.days_ago(0),    
+    default_args={ 'on_failure_callback': on_failure_callback}
+) as crm_workflow_dag:
+
+    validate_file_exists = GCSObjectExistenceSensor(
+        task_id="validate_file_exists",
+        bucket=BUCKET_NAME,
+        object=FILE_NAME
+    )
+
+    start_dataflow_job = DataflowTemplatedJobStartOperator(
```

Comment:
Just double checking that using this template automatically kicks off a batch job right? (asking because I've 100% accidentally used Airflow to kick off streaming pipelines that I then later forgot to stop and ran for months without me realizing)

Reply:
Ha yes! It's just a batch word count Dataflow job using a Dataflow template